ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala

```diff
@@ -172,22 +202,29 @@ class CommandService(
     : Error \/ lav1.value.Value = {
     val result: Option[lav1.value.Value] = for {
       transaction <- a.transaction: Option[lav1.transaction.TransactionTree]
-      treeEvent <- rootTreeEvent(transaction): Option[lav1.transaction.TreeEvent]
-      exercised <- treeEvent.kind.exercised: Option[lav1.event.ExercisedEvent]
+      exercised <- firstExercisedEvent(transaction): Option[lav1.event.ExercisedEvent]
       exResult <- exercised.exerciseResult: Option[lav1.value.Value]
     } yield exResult
 
     result.toRightDisjunction(
       Error(
         'choiceArgument,
-        s"Cannot get exerciseResult from the first root event of gRPC response: ${a.toString}"))
+        s"Cannot get exerciseResult from the first ExercisedEvent of gRPC response: ${a.toString}"))
   }
 
-  private def rootTreeEvent(
-      a: lav1.transaction.TransactionTree): Option[lav1.transaction.TreeEvent] =
-    a.rootEventIds.headOption.flatMap { id =>
-      a.eventsById.get(id)
+  private def firstExercisedEvent(
+      tx: lav1.transaction.TransactionTree
+  ): Option[lav1.event.ExercisedEvent] = {
+    @tailrec
+    def loop(ids: Seq[String]): Option[lav1.event.ExercisedEvent] = ids match {
+      case h +: t =>
+        val result = tx.eventsById.get(h).flatMap(_.kind.exercised)
+        if (result.isDefined) result
+        else loop(t)
+      case Nil => None
     }
+    loop(tx.rootEventIds)
```

Comment:
```suggestion
    tx.rootEventIds.collectFirst(Function unlift { h =>
      ...
```

Also, `case Nil` is unsafe; using `collectFirst` avoids this problem, too.

Reply:
ummmm. What is unsafe about `case Nil` and why `scalac` does not tell me about this :)

Reply:
is that because it is `case object Nil extends List[Nothing]`?